src/core/DY.Core.Trace.Manipulation.fst

```diff
@@ -299,18 +299,18 @@ let mk_rand_get_label #invs usg lab len tr =
 
 /// Usage of random bytestrings.
 
-val mk_rand_get_usage:
+val mk_rand_has_usage:
   {|protocol_invariants|} ->
   usg:usage -> lab:label -> len:nat{len <> 0} -> tr:trace ->
   Lemma
   (ensures (
     let (b, tr_out) = mk_rand usg lab len tr in
-    get_usage b == usg
+    b `has_usage tr_out` usg
```

Comment:
This is definitely a weaker guarantee, so the immediate question is if we lose anything from this. It seems that in many cases, `has_usage` is all we need, but I wonder if there's some way to retain the stronger guarantee in places where we have that information, and then just weaken it to `has_usage` if the full strength isn't needed.

Reply:
Indeed, `has_usage` is weaker than using `get_usage`, hence using it is weakening post-conditions (bad) but is also weakening pre-conditions (good), so it's more a different compromise on DY* as a whole.

Here, it's weakening because of the "or is publishable" baked-in `has_usage`, but hopefully we can reason on publishability separately because `mk_rand` offers strong guarantees on the label.
https://github.com/REPROSEC/dolev-yao-star-extrinsic/blob/60ddb19cc7ab3ea9d27121d42e617411b28eef0e/src/core/DY.Core.Trace.Manipulation.fst#L290-L294

Reply:
I added the stronger variant of the lemma in `mk_rand_get_usage`